roles/download/tasks/main.yml

```diff
@@ -26,5 +59,29 @@
     - not skip_downloads | default(false)
     - download.enabled
     - item.value.enabled
```

Comment:
I would split this in two tasks (download_container / download_file) to simplify the loops and condition, see below